aten/src/ATen/function_wrapper.py

```diff
@@ -173,7 +177,7 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 # the same name (but different signature) already
 ZERO_DIM_CHECK = CodeTemplate("""\
 if (${check_name}.dim() == 0) {
-    return static_cast<const Type*>(this)->${api_name}(${zero_dim_actuals});
+    return static_cast<const TypeInternalInterface*>(this)->${api_name}(${zero_dim_actuals});
```

Comment:
I'm not really a fan of TypeInternalInterface, because we really consider all of Type to be internal (I guess it's a question of whether internal means 'internal to C++ users' or 'internal to Tensor'.  What's wrong with TypeFunctions or similar?

Reply:
or maybe 'Private' instead of Internal?

Reply:
Maybe, `TypeExtendedInterface`?

Reply:
I don't like `TypeFunctionInterface` because it doesn't make clear that you can also get the method functions from the interface.

Reply:
TypeExtendedInterface sounds good.